networkx/classes/function.py

```diff
@@ -793,6 +793,23 @@ def set_edge_attributes(G, values, name=None):
         >>> (1, 2) in G.edges()
         False
 
+    Note for multigraphs , values argument should be dicts with 3-tuples key
+    instead of 2-tuples key commonly used in non-multigraphs. The edge key has
+    to be passed as the third value in the tuple::
+
+        >>> import networkx as nx
+        >>> G = nx.MultiDiGraph()
+        >>> edges = [(0,1), (0,1), (1,0), (2,2), (3, 4), (1,3)]
+        >>> G.add_edges_from(edges)
+        [0, 1, 0, 0, 0, 0]
+        >>>
+        >>> attributes = {(0,1,0): {"weight": 20, "cost": 21}, (2,2,4): {"weight": 10, "cost": 3}, (0,1,1): {"cost":7}}
+        >>> nx.set_edge_attributes(G, attributes)
+        >>> G[0][1][0]["cost"]
+        21
+        >>> G[0][1][1]["cost"]
+        7
```

Comment:
As a general note, the example should be as minimal as possible and focus on highlighting the single feature described. In this case, that feature is the 3-tuple edge keys, so the extra non-multiedges are not really important to the example. I think the example would be improved by removing them and focusing only on the mutliple `(0, 1)` edges, e.g. something like:

```suggestion
        >>> G = nx.MultiGraph()
        >>> edges = [(0,1), (0,1)]
        >>> G.add_edges_from(edges)  # Returns list of edge keys
        [0, 1]
        >>> attributes = {(0,1,0): {"cost": 21}, (0,1,1): {"cost": 7}}
        >>> nx.set_edge_attributes(G, attributes)
        >>> G[0][1][0]["cost"]
        21
        >>> G[0][1][1]["cost"]
        7
```